include/simulation_time/simulation_clock.hpp

```diff
@@ -0,0 +1,219 @@
+#pragma once
+
+#include <cassert>
+#include <chrono>
+#include <stdexcept>
+#include <utility>
+
+namespace ngen {
+
+//! Simulation Clock
+//!
+//! Represents simulation ticks monotonically increasing from zero, and
+//! provides conversion utilities between simulation ticks and
+//! calendar time.
+//!
+//! Implements named requirements: Clock (https://en.cppreference.com/w/cpp/named_req/Clock)
+//!
+struct simulation_clock final
+{
+    /* Clock named requirements */
+
+    using rep = std::int64_t; // TODO: double vs std::int64_t
```

Comment:
I don't think we're likely to run into sub-second time precision, so `int64_t` is probably fine.